src/Microsoft.Extensions.Configuration.AzureAppConfiguration/SrvLookupClient.cs

```diff
@@ -0,0 +1,130 @@
+
+            _tcpLookupClient = new LookupClient(new LookupClientOptions()
+            {
+                UseTcpOnly = true
+            });
+        }
+
+        public async Task<IEnumerable<SrvRecord>> QueryAsync(string host, CancellationToken cancellationToken)
+        {
+            IEnumerable<SrvRecord> resultRecords = Enumerable.Empty<SrvRecord>();
+            var originSrvDns = $"{TcpOrigin}.{host}";
+
+            bool exists = _cachedOriginHosts.TryGetValue(originSrvDns, out OriginHostCacheItem originHost);
+
+            if (!exists || originHost.CacheExpires <= DateTimeOffset.UtcNow)
+            {
+                IEnumerable<SrvRecord> records = await InternalQueryAsync(originSrvDns, cancellationToken).ConfigureAwait(false);
+
+                if (records == null || records.Count() == 0)
+                {
+                    return Enumerable.Empty<SrvRecord>();
+                }
+
+                if (!exists)
+                {
+                    originHost = new OriginHostCacheItem()
+                    {
+                        OriginHost = records.First().Target.Value.Trim('.'),
+                        CacheExpires = DateTimeOffset.UtcNow.Add(FailOverConstants.OriginHostResultCacheExpiration)
+                    };
+
+                    _cachedOriginHosts[originSrvDns] = originHost;
+                }
+                else
+                {
+                    originHost.OriginHost = records.First().Target.Value.Trim('.');
+                    originHost.CacheExpires = DateTimeOffset.UtcNow.Add(FailOverConstants.OriginHostResultCacheExpiration);
+                }
+            }
+
+            int index = 0;
+
+            while (true)
+            {
+                string altSrvDns = $"{Alt}{index}.{TCP}.{originHost.OriginHost}";
+
+                IEnumerable<SrvRecord> records = await InternalQueryAsync(altSrvDns, cancellationToken).ConfigureAwait(false);
+
+                if (records == null)
+                {
+                    break;
+                }
+                resultRecords = resultRecords.Concat(records);
+
+                // If we get less than 20 records from _alt{i} SRV, we have reached the end of _alt* list
+                if (records.Count() < MaxSrvRecordCountPerRecordSet)
+                {
+                    break;
+                }
+
+                index++;
+            }
+
+            return resultRecords;
+        }
+
+        private async Task<IEnumerable<SrvRecord>> InternalQueryAsync(string srvDns, CancellationToken cancellationToken)
+        {
+            var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            cts.CancelAfter(FailOverConstants.UdpSrvQueryTimeout);
+
+            IDnsQueryResponse dnsResponse;
+
+            try
+            {
+                dnsResponse = await _udpLookupClient.QueryAsync(srvDns, QueryType.SRV, QueryClass.IN, cts.Token).ConfigureAwait(false);
+            }
+            catch (Exception e) when

```

Comment:
Do you mind to update this to avoid `catch Exception` appearing. Catching base exception is not an appropriate approach, and although here we filter the exception by type, the catch statement itself is already good for that.